bundles/org.openhab.binding.nikohomecontrol/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -624,6 +786,102 @@
+		<tags>
+			<tag>Status</tag>
+		</tags>
+		<state readOnly="true">
+			<options>
+				<option value="ACTIVE">@text/channelOptionChargingStatusActive</option>
+				<option value="INACTIVE">@text/channelOptionChargingStatusInactive</option>
+				<option value="BATTERY FULL">@text/channelOptionChargingStatusBatteryFull</option>
+				<option value="ERROR">@text/channelChargingStatusError</option>
+			</options>
+		</state>
+	</channel-type>
+	<channel-type id="evStatus">
+		<item-type>String</item-type>
+		<label>@text/channelEvStatusLabel</label>
+		<description>@text/channelEvStatusDescription</description>
+		<tags>
+			<tag>Status</tag>
+		</tags>
+		<state readOnly="true">
+			<options>
+				<option value="IDLE">@text/channelOptionEvStatusIdle</option>
+				<option value="CONNECTED">@text/channelOptionEvStatusConnected</option>
+				<option value="CHARGING">@text/channelOptionEvStatusCharging</option>
+			</options>
+		</state>
+	</channel-type>
+	<channel-type id="couplingStatus">
+		<item-type>String</item-type>
+		<label>@text/channelCouplingStatusLabel</label>
+		<description>@text/channelCouplingStatusDescription</description>
+		<tags>
+			<tag>Status</tag>
+		</tags>
+		<state readOnly="true">
+			<options>
+				<option value="OK">@text/channelOptionCouplingStatusOk</option>
+				<option value="NO INTERNET">@text/channelOptionCouplingStatusNoInternet</option>
+				<option value="NO CREDENTIALS">@text/channelOptionCouplingStatusNoCredentials</option>
+				<option value="INVALID CREDENTIALS">@text/channelOptionCouplingStatusInvalidCredentials</option>
+				<option value="CONNECTION ERROR">@text/channelOptionCouplingStatusConnectionError</option>
+				<option value="CONNECTION TIMEOUT">@text/channelOptionCouplingStatusConnectionTimeout</option>
+				<option value="API ERROR">@text/channelOptionCouplingStatusApiError</option>
+				<option value="UNKNOWN ERROR">@text/channelOptionCouplingStatusUnknownError</option>
+			</options>
+		</state>
+	</channel-type>
+	<channel-type id="chargingMode">
+		<item-type>String</item-type>
+		<label>@text/channelChargingModeLabel</label>
+		<description>@text/channelChargingModeDescription</description>
+		<tags>
+			<tag>Control</tag>
+		</tags>
+		<state>
+			<options>
+				<option value="SOLAR">@text/channelOptionChargingModeSolar</option>
+				<option value="NORMAL">@text/channelOptionChargingModeNormal</option>
+				<option value="SMART">@text/channelOptionChargingModeSmart</option>
+			</options>
+		</state>
+	</channel-type>
+	<channel-type id="targetDistance">
+		<item-type unitHint="km">Number:Length</item-type>
+		<label>@text/channelTargetDistanceLabel</label>
+		<description>@text/channelTargetDistanceDescription</description>
+		<state pattern="%d %unit%"/>
+	</channel-type>
+	<channel-type id="targetTime">
+		<item-type>DateTime</item-type>
+		<label>@text/channelTargetTimeLabel</label>
+		<description>@text/channelTargetTimeDescription</description>
+		<state pattern="%tR"/>
```

Comment:
Can you add tags to most if not all channel / types ?

Reply:
Done